provider/adapters/src/fallback/adapter.rs

```diff
@@ -129,7 +131,7 @@ impl<P> LocaleFallbackProvider<P> {
     {
         let key_fallbacker = self.fallbacker.for_key(key);
         let mut fallback_iterator = key_fallbacker.fallback_for(base_req.clone());
-        while !fallback_iterator.get().options.is_empty() {
+        loop {
```

Comment:
This is a bug fix; if the incoming locale starts at `und` we still need to check for it in the provider.